packages/function-tree/index.d.ts

```diff
@@ -0,0 +1,141 @@
+    functionIndex: number
+    items: Array<FunctionTreePrimitive>
+    type: "parallel" | "sequence"
+    _functionTreePrimitive: boolean
+    outputs?: { [name: string]: FunctionTreePrimitive }
+}
+
+export class Sequence {
+    constructor(items: Array<FunctionTreePrimitive>)
+    constructor(name: string, items: Array<FunctionTreePrimitive>)
+
+    toJSON(): FunctionTreePrimitive
+}
+
+export class Parallel {
+    constructor(items: Array<FunctionTreePrimitive>)
+    constructor(name: string, items: Array<FunctionTreePrimitive>)
+
+    toJSON(): FunctionTreePrimitive
+}
+
+export function sequence(items: Array<FunctionTreePrimitive>): Sequence
+export function sequence(name: string, items: Array<FunctionTreePrimitive>): Sequence
+
+export function parallel(items: Array<FunctionTreePrimitive>): Parallel
+export function parallel(name: string, items: Array<FunctionTreePrimitive>): Parallel
+
+export interface RunTreeFunction {
+    (): void
+    on(event: string | symbol, listener: Function): this
+    once(event: string | symbol, listener: Function): this
+    off(event: string | symbol, listener: Function): this
+}
+
+
+export interface DebounceFunction {
+    displayName: string
+    // What Promise returns ?
+    (params: { path: string }): Promise<any>
+}
+
+export interface DebounceFactory {
+    (time: number): DebounceFunction
+    shared(): (time: number) => DebounceFunction
+}
+
+export interface DevtoolsOptions {
+    remoteDebugger: boolean
+}
+
+export class Devtools {
+    constructor(options: DevtoolsOptions)
+
+    addListeners(): void
+    init(): void
+    // Should there be object or something else ? Maybe '{}' ?
+    safeStringify(object: object): string
+    reInit(): void
+    sendMessage(message: string): void
+    watchExecution(tree: FunctionTree): void
+    sendInitial(): void
+    // What shape debugggingData and context has ?
+    createExecutionMessage(debuggingData: object, context: any, functionDetails: FunctionTreePrimitive, payload: IPayload): string
+    sendExecutionData(debuggingData: object, context: any, functionDetails: FunctionTreePrimitive, payload: IPayload): void
+    Provider(): (context: any, functionDetails: FunctionTreePrimitive, payload: IPayload) => any
+}
+
+
+export class FunctionTreeError extends Error {
+    constructor(error: any)
+    toJSON(): any
+}
+
+export class FunctionTreeExecutionError extends FunctionTreeError {
+    constructor(execution: any, funcDetails: any, payload: any, error: any)
+    toJSON(): any
+}
+
+export type Provider = (context: any, funcDetails: FunctionTreePrimitive, payload: IPayload, next: IPayload) => any
+
+export function ReduxProvider(store: any): Provider
+export function PropsProvider(): Provider
+export function PathProvider(): Provider
+export function ExecutionProvider(execution: any, abort: Abort): Provider
+export function ContextProvider(extendedContext: any): Provider
+
+export class FunctionTree extends NodeJS.EventEmitter {
```

Comment:
we extend eventemitter3. which seems to [have own typings](https://github.com/primus/eventemitter3/blob/master/index.d.ts) why we declare here NodeJS.EventEmitter?